### DISCUSSION TITLE
Order of FieldContainer's data

### DISCUSSION THREAD

**Discussion by PanekOndrej** [ansys/pydpf-core#322] [Q&A] [answered]
Hi everyone,

I found out, that FieldsContaintainer's data are in different order compared to MAPDL printouts. Is there any ellegant way how to sort the **ElementalNodal results** in the same order as Mechanical (e.g. `PRESOL, S, COMP` that gives for 20-node solid results in order Element1: Components in nodes I-P, Element2: Components in nodes I-P ... )?

  ELEMENT=       1        SOLID186
    NODE     SX           SY           SZ           SXY          SYZ          SXZ     
       1   1.3263       284.93      -39.084      -16.892      -43.007       4.0296     
     262   21.590       351.41      -6.1419      -25.433      -5.2396       1.8689     
     323   21.515       386.07      -9.9333      -3.8139     -0.21751       3.6931     
     325   16.530       327.33      -16.098      -2.5938     -0.23076       13.250     
     114  -2.6839       254.75      -36.265      -14.772      -91.788       4.9086     
     318   15.213       348.90      -1.8150      -24.207      -25.527       5.0770     
     322   2.2081       370.61      -14.374      -5.8494      -8.7611       1.0131     
     329   4.8359       317.03      -21.273      -3.3133      -14.164       5.8482     
  
  ELEMENT=       2        SOLID186
    NODE     SX           SY           SZ           SXY          SYZ          SXZ     
       1   11.011       294.04      -36.009      -12.592      -33.715       12.915     
     325   38.955       341.07      -6.8376      -4.9140       1.0630       17.036     
     323   34.593       390.75      -10.829      -4.5121      0.30133       1.1660     
     262   27.949       354.76      -7.7270      -14.852      -2.7532       1.7589     
       2   30.410       310.29      -24.320      -1.1019      -39.836       4.8410     
     327   43.250       336.58      -17.072     -0.67958     -0.99439       3.8806     
     324   40.006       385.74      -19.990      0.12635      -2.7101       5.0058     
     263   58.894       386.73       18.833       4.3321      -3.5496      0.38066
     
and **averaged results** (e.g. `PRNSOL, S, COMP`)?
 
 NODE     SX           SY           SZ           SXY          SYZ          SXZ     
       1  -11.914       128.40      -28.133      -10.570      -75.435       3.1811     
       2  -14.178       117.61      -35.533       6.7231      -83.941       6.8971     
       3   28.375      -301.34       24.229       17.638      -36.117       4.4236     
       4   33.115      -289.88       35.071      -15.315      -36.145       1.1863     
       5  -1.0247       35.169       5.9328     -0.85126      -72.325      -6.3782     
       6   1.6989       36.430       9.7768       1.8942      -72.843       6.8301     
       7 -0.84983      -169.72      -4.8835      -3.3034      -51.890      -4.8353     
       8  -8.6473      -183.54      -16.923       4.3810      -53.359       4.3550     
       9  0.43478      -34.304      -7.0810     -0.45373      -51.191     -0.90355     
      10 -0.32724E-001 -33.986      -8.5274      -1.1628      -49.457     -0.21620     

Best regards,
OP

**Comment by rlagha** [comment, marked as answer]
@PProfizi please could you provide an example using the sort operator? @PanekOndrej , please let us know if this is your request?

**Reply by PanekOndrej** [reply]
Yes, I couldn't find any documentation about "sort" operator before. It seems that "ascending_sort" operator, that will be available in new version of ansys.dpf.core package (>= 0.5.0) will solve my problem, right? ("ascending_sort" raises AttributeError in 0.4.2)

**Reply by PProfizi** [reply]
Hi @PanekOndrej , 
To sort the nodal averaged output of an ElementalNodal field according to a sequence of node IDs, you can indeed use the sorting operator using a nodal scoping (such as the element connectivity) like so:
```
from ansys.dpf import core as dpf
from ansys.dpf.core import examples
from ansys.dpf.core.operators.logic import ascending_sort

model = dpf.Model(examples.download_piston_rod())

# You can first define an ordered list of elements of interest
elemental_scoping = dpf.mesh_scoping_factory.elemental_scoping([1, 2])
# You can then potentially get the ordered list of node IDs for these elements
connectivity_ids = dpf.operators.scoping.connectivity_ids(mesh_scoping=mesh_scoping,
                                                          mesh=model.metadata.meshed_region).eval()
# You can then extract the nodal averaged stress values as a field, on this nodal scoping
nodal_stress_field = model.results.stress.on_mesh_scoping(mesh_scoping=connectivity_ids).eval()[0]
# To force the data to follow the scoping order, you can then use an ordering operator such as:
sort_nodal_by_connectivity_scoping = ascending_sort(field=nodal_stress_field,
                                                    sort_by_scoping=True).eval()
# When printing out the data, it is ordered according to the nodal scoping used 
# (here connectivity, but could be a custom scoping)
print(sort_nodal_by_connectivity_scoping.data)
```
The `ansys.dpf.core.operators.scoping.rescope` operator should also be able to force the reordering of your data according to a scoping of your choice.

Does this answer your question?
